receiver/sqlqueryreceiver/metrics.go

```diff
@@ -46,6 +50,20 @@ func rowToMetric(row metricRow, cfg MetricCfg, dest pmetric.Metric) error {
 	return nil
 }
 
+func setTimestamp(cfg MetricCfg, dp pmetric.NumberDataPoint, ts pcommon.Timestamp, scrapeCfg scraperhelper.ScraperControllerSettings) {
+	dp.SetTimestamp(ts)
+
+	switch cfg.Aggregation {
+	// Cumulative sum should have a start time set to the beginning of the data points cumulation
+	case MetricAggregationCumulative:
+		dp.SetStartTimestamp(ts)
+
+	// Non-cumulative sum should have a start time set to the previous endpoint
+	case MetricAggregationDelta:
+		dp.SetStartTimestamp(pcommon.NewTimestampFromTime(ts.AsTime().Add(-scrapeCfg.CollectionInterval)))
+	}
```

Comment:
And this should only be applied to sums, not gauges

Reply:
submitted change to ensure start time only gets added to count metrics .